### PR TITLE
messagecenter topic ui

### LINKED ISSUE #1182: Message Center UI
Message Center is a UI component that allow to view and send messages
UI will show maximum 2 topics in message center.

- [x] **Add meta/messageCenter:**
```
enum TopicStatus = {
opened ='opened'
resolved='resolved'
}
interface Message {
    id: number
}

interface Topic {
  id:number
  key:string
  assessmentId: number
  countryIso: string
  cycleId?: number
  title?: string
  status?:TopicStatus
  messages?:Array<Message>
}
```

- [x] **Add store ui/messageCenter** 
```
messageCenter : {
    topics: Array<Topic>
}
```

- [x] **export hooks from ui/messageCenter**
```
useTopics():Array<Topic>
```

- [x] **export MessageCenterActions from ui/messageCenter**
```
MessageCenterActions {
open({topic:Topic}) // adds a topic to the queue
// maximum 2 topic can be in the queue
}
```

- [x] **Add component components/MessageCenter**

Ui Topic component is very very similar to current fra-review__container.

Difference with fra-review__container:
**fra-review__container** :
    position: fixed;
    top: 64px;
    right: 32px; OR calc(32px + 32px + 288px)
    height: calc(100% - 64px);
    box-shadow: 0px 0px 12px 11px #dadada;
**fra-review__header**
remove min-height and remove sub_header element
**fra-review__footer**
show only add button and aligned to the right

**IMPORTANT:**
Add Only header, empty content message, footer

**SEE MOCKUP:**
    
Screenshot 2022-03-04 at 15 32 53

### PR DESCRIPTION
**Pull request by rfontanarosa**
closes #1182 
Schermata 2022-03-22 alle 16 26 21

### CODE REVIEW

File: client/components/MessageCenter/MessageCenter.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        </div>
+      </div>
+    </div>
+  )
+}
+
+const MessageCenter: React.FC = () => {
+  const topics = useTopics()
+
+  if (!topics) return null
```
Comment (minotogna) [inline client/components/MessageCenter/MessageCenter.tsx:30]: this won't work if your initial state is an empty array. check if it's empty ?

File: client/store/ui/messageCenter/slice.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  topics: Array<MessageTopic>(),
+}
+
+export const messageCenterSlice = createSlice({
+  name: 'messageCenter',
+  initialState,
+  reducers: {
+    reset: () => initialState,
+    open: (state, action) => {
+      if (state.topics.length < 2) state.topics.push(action.payload)
```
Comment (minotogna) [inline client/store/ui/messageCenter/slice.ts:14]: it should always be added.
the first entered should be removed !

File: client/components/MessageCenter/MessageCenter.tsx (hunk 1 above)
Comment (rfontanarosa) [reply client/components/MessageCenter/MessageCenter.tsx:30]: import { Objects } from '@core/utils'

File: client/store/ui/messageCenter/slice.ts (hunk 2 above)
Comment (rfontanarosa) [reply client/store/ui/messageCenter/slice.ts:14]: if (state.topics.length === 2) state.topics.shift()
      state.topics.push(action.payload)

File: client/components/MessageCenter/MessageCenter.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        {topic.key}
+        <div className="topic-close" onClick={() => null} onKeyDown={() => null} role="button" tabIndex={0}>
+          <Icon name="remove" />
+        </div>
+      </div>
+      <div className="topic-body">
+        <div className="no-comments">
+          <Icon className="icon-24" name="chat-46" />
+          <br />
+          No comments
```
Comment (minotogna) [inline client/components/MessageCenter/MessageCenter.tsx:25]: use i18n

File: client/components/MessageCenter/MessageCenter.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import { useTopics } from '@client/store/ui/messageCenter'
+import Icon from '@client/components/Icon'
+import { Objects } from '@core/utils'
+import { MessageTopic } from '@meta/messageCenter'
+
+type TopicProps = {
+  topic: MessageTopic
+}
+
+const Topic: React.FC<TopicProps> = ({ topic }) => {
```
Comment (minotogna) [inline client/components/MessageCenter/MessageCenter.tsx:12]: export Topic to its own file

**Review by minotogna — CHANGES_REQUESTED** [review]

File: client/components/MessageCenter/MessageCenter.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        <div className="no-comments">
+          <Icon className="icon-24" name="chat-46" />
+          <br />
+          No comments
+        </div>
+      </div>
+      <div className="topic-footer">
+        <textarea />
+        <button className="btn-s btn-primary" type="submit">
+          Add
```
Comment (minotogna) [inline client/components/MessageCenter/MessageCenter.tsx:31]: Use i18n

**Review by minotogna — APPROVED** [review]

**Pull request merged**